### PR TITLE
Watchdog edits for 19-2.1

### PR DESCRIPTION
**Pull request by papampi**
Watchdog edits:
Remove echo status nvidia-smi, add an echo after check each GPU
Raise nvidia-smi query each GPU from 0.2 sec to 0.5 sec 
Raise number of loops to go before restart 3main if all GPUs below threshold to give miner more time to correct itself.

### CODE REVIEW

**Comment by papampi** [comment]
Any more fixes ?
Both watchdog and tempcontrol have calls for nvidia-smi, is it possible to make one read from another one?

**Review by nvOC-Stubo — COMMENTED** [review]
I think I may understand part of the problem being addressed here. That is that the watchdog is sometimes to quick to restart the miner and you would like to give it more time to recover on its own. This makes sense but keep in mind that there is a careful balance that must be made with this change as the effect for systems with a large number of GPUs vs. 1 or 2 is very different.

That being said, I see you remove the call to nvidia-smi on line 186 but I don't see it anywhere else. Of course, that call is used to set the UTILIZATIONS array and is required for the watchdog to work properly.

**Comment by papampi** [comment]
@nvOC-Stubo Line 186 call was only to echo the old style GPU utilization I remember you removed it and add it again when me and leenoox asked for the utilization echo.
New echo style added here:
https://github.com/papampi/nvOC_by_fullzero_Community_Release/blob/70d5dd8fd350f6d8ae6fe66ac336d017e67fc61e/5watchdog#L209

**Comment by papampi** [comment]
About too quick restart of 3main, I see many times on both rigs with 12 cards and my test rig with 3 cards that pool gave timeout and miner is re-connecting but watchdog not giving enough time to reconnect and restart 3main.
We also have the no miner 3main restart, so if the error is with cards usually miner quits and watchdog will restart 3main as fast as it catch no miner.
And with latest 3main edits we have a lot faster 3main restart too

**Comment by nvOC-Stubo** [comment]
I show line 187 as the echo which is also removed but it looks like part of the edit is to also remove line 186:
-  UTILIZATIONS=$(nvidia-smi --query-gpu=utilization.gpu --format=csv,noheader,nounits)

Is that not the case? UTILIZATIONS must be set somewhere.

**Comment by papampi** [comment]
Lines 185-188 were only for echoing the UTILIZATION in old watchdog style no calls for the UTILIZATION anywhere else in your last watchdog edit.

https://github.com/papampi/nvOC_by_fullzero_Community_Release/blob/1cb8b93198683a42194c94938055b96d618549fc/5watchdog#L185-L188

All the checks use your new $UTIL from:
https://github.com/papampi/nvOC_by_fullzero_Community_Release/blob/1cb8b93198683a42194c94938055b96d618549fc/5watchdog#L213

**Comment by nvOC-Stubo** [comment]
Ah, got it. That code was put there to spit out a list of all GPU utilizations in one big chunk so it is redundant and can be removed. Another thing to consider is making the sleep time at the bottom of the script (10 or 15) a variable in 1bash so that users can configure it without the daunting task of editing the watchdog code.

**Comment by papampi** [comment]
No one has any edits/additions?
Merge?

File: 5watchdog [hunk 1]
```diff
@@ -61,6 +61,7 @@ echo "Watchdog for $nvOC_Ver"
 echo "Version: $nvOC_wdog_ver"
 echo ""
 
+export NVOC="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
Comment (LuKePicci) [inline 5watchdog]: I would remove this line for the same reasoning discussed elsewhere, but take this as good general guideline and nothing more.

**Pull request merged**